include/cucumber-cpp/internal/CukeDll.hpp

```diff
@@ -0,0 +1,14 @@
+#ifndef CUKE_CUKEDLL_HPP_
+#define CUKE_CUKEDLL_HPP_
+
+#ifndef CUKE_API_
+#if CUKE_LINKED_AS_SHARED_LIBRARY
```

Comment:
Just like GenerateExportHeader, please:

* wrap both the export and import case in `#ifndef CUCUMBER_CPP_STATIC_DEFINE` to detect when we're building or using a static library.
* distinguish the export from the import case by the presence of the macros `cucumber_cpp_EXPORTS` and `cucumber_cpp_nomain_EXPORTS` (CMake defines those only while building objects for the mentioned target)